File: v131016/TeeNew/Series_DeltaPoint.pas
unit Series_DeltaPoint;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  ExtDlgs,
  {$ENDIF}
  Base, TeEngine, Series, ImaPoint, TeeProcs, Chart;

type
  TDeltaPointForm = class(TBaseForm)
    Series1: TDeltaPointSeries;
    Label1: TLabel;
    Image1: TImage;
    Image2: TImage;
    Image3: TImage;
    procedure FormCreate(Sender: TObject);
    procedure Image1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

uses TeeBrushDlg;

procedure TDeltaPointForm.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(10);

  Image1.Picture.Assign(Series1.LowerImage);
  Image2.Picture.Assign(Series1.EqualImage);
  Image3.Picture.Assign(Series1.GreaterImage);
end;

procedure TDeltaPointForm.Image1Click(Sender: TObject);
var tmp : TImage;
    tmpSt : String;
begin
  tmpSt:=TeeGetPictureFileName(Self);
  if tmpSt<>'' then
  begin
    tmp:=(Sender as TImage);
    tmp.Picture.LoadFromFile(tmpSt);
    if tmp=Image1 then Series1.LowerImage.Assign(tmp.Picture)
    else
    if tmp=Image2 then Series1.EqualImage.Assign(tmp.Picture)
    else
    if tmp=Image3 then Series1.GreaterImage.Assign(tmp.Picture);
  end;
end;

initialization
  RegisterClass(TDeltaPointForm);
end.

File: TeeNew/Builder2007/Series_DeltaPoint.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Series_DeltaPoint.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "ImaPoint"
#pragma resource "*.dfm"
TDeltaPointForm *DeltaPointForm;
//---------------------------------------------------------------------------
__fastcall TDeltaPointForm::TDeltaPointForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TDeltaPointForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(10);

  Image1->Picture->Assign(Series1->LowerImage);
  Image2->Picture->Assign(Series1->EqualImage);
  Image3->Picture->Assign(Series1->GreaterImage);
}
//---------------------------------------------------------------------------

void __fastcall TDeltaPointForm::Image1Click(TObject *Sender)
{
  if (OpenPictureDialog1->Execute())
  {
    TImage *tmp = dynamic_cast<TImage*>(Sender);
    tmp->Picture->LoadFromFile(OpenPictureDialog1->FileName);
    if (tmp == Image1) Series1->LowerImage->Assign(tmp->Picture);
    else if (tmp == Image2) Series1->EqualImage->Assign(tmp->Picture);
    else if (tmp == Image3) Series1->GreaterImage->Assign(tmp->Picture);
  }
}
//---------------------------------------------------------------------------

